kube-runtime/src/reflector/mod.rs

```diff
@@ -97,7 +100,19 @@ where
     K::DynamicType: Eq + Hash + Clone,
     W: Stream<Item = watcher::Result<watcher::Event<K>>>,
 {
-    stream.inspect_ok(move |event| writer.apply_watcher_event(event))
+    let mut stream = Box::pin(stream);
+    stream! {
+        while let Some(event) = stream.next().await {
+            match event {
+                Ok(ev) => {
+                    writer.apply_watcher_event(&ev);
+                    writer.dispatch_event(&ev).await;
+                    yield Ok(ev);
+                },
+                Err(ev) => yield Err(ev)
+            }
+        }
+    }
```

Comment:
i feel like this does not need to use `stream!` and should work with [`StreamExt::then`](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.then).

Reply:
hmm this feels like it'd be a good change but not sure how it would work, since the closure passed to `then()` would capture the writer. We need a mutable reference to the writer, and we need one for each element of the stream. The compiler shouted at me when I tried.